Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -147,13 +149,32 @@ class BrowserViewController: UIViewController {
         
         #if NO_REWARDS
         rewards = nil
+        rewardsObserver = nil
         #else
         RewardsHelper.configureRewardsLogs()
         rewards = BraveRewards(configuration: .default)
+        rewardsObserver = LedgerObserver(ledger: rewards!.ledger)
+        rewards?.ledger.add(rewardsObserver!)
         #endif
 
         super.init(nibName: nil, bundle: nil)
         didInit()
+        
+        #if !NO_REWARDS
```

Comment:
Can we extract it to `rewardsObserverSetup` method?

Reply:
`rewardsObserverSetup`? If you mean the code above super.init: unfortunately not because  we reference self in the closures

Reply:
Below init, just moving these `fetchedPanelPublisher` and rest to a new separate method to not flood the initializer with code
Sorry for not being clear

Reply:
Ah, sure yeah I can do that